src/RulesEngine/RulesEngine.cs

```diff
@@ -149,11 +149,37 @@ private async ValueTask<ActionRuleResult> ExecuteActionForRuleResult(RuleResultT
         #region Private Methods
 
         /// <summary>
-        /// Adds the workflow.
+        /// Adds the workflow if the workflow name is not already added. Ignores the rest.
         /// </summary>
         /// <param name="workflowRules">The workflow rules.</param>
         /// <exception cref="RuleValidationException"></exception>
         public void AddWorkflow(params WorkflowRules[] workflowRules)
+        {
+            try
+            {
+                foreach (var workflowRule in workflowRules)
+                {
+                    if (!_rulesCache.ContainsWorkflowRules(workflowRule.WorkflowName))
```

Comment:
I think we should validate the workflow first before checking it exists.
WorkflowName can be null if input is incorrect